protocol.go

```diff
@@ -99,6 +99,9 @@ type protocolHandler interface {
 	// request's context, a nil cancellation function, and a nil error.
 	SetTimeout(*http.Request) (context.Context, context.CancelFunc, error)
 
+	// CanHandlePayload returns true if the protocol can handle an HTTP request.
```

Comment:
nit: In the docs, can you mention that this is called _after_ we've already validated that the HTTP method is supported?

Reply:
Oops, done in 5b79b761420b1460854696c6ce2ad9901545f9f7.